trio/tests/test_util.py

```diff
@@ -310,3 +310,18 @@ def test_bad_pathlike_implementation(self, exception, method):
         klass.__fspath__ = method
         with pytest.raises(exception):
             fspath(klass())
+
+
+def test_is_main_thread():
+    _thread_result = None
+
+    def _():
+        nonlocal _thread_result
+        _thread_result = is_main_thread()
+
+    assert is_main_thread()
+    thread = threading.Thread(target=_)
+    thread.start()
+    thread.join()
+    assert _thread_result is not None
+    assert _thread_result is False
```

Comment:
Can you write this as

```python
async def test_is_main_thread():
    assert is_main_thread()

    async def not_main_thread():
        assert not is_main_thread()

    await run_sync_in_worker_thread(not_main_thread)
```

(I keep forgetting this is an option too and having to remind myself :-). But it really is a better habit I think -- not just because it's simpler, but e.g. in the test as originally written if there was an exception inside `_` then it would get lost.)